functional_testing/Openvcloud/RESTful/testcases/libcloud/test01_libvirt.py

```diff
@@ -0,0 +1,151 @@
+import time, random
+from testcases import *
+from nose_parameterized import parameterized
+
+class LibCloudBasicTests(TestcasesBase):
+
+    def setUp(self):
+        super().setUp()   
+        response = self.api.cloudapi.locations.list()
+        self.assertEqual(response.status_code, 200)
+
+        locations = response.json()
+        if not locations:
+            self.skipTest('No locations were found in the environment')
+        
+        self._location = random.choice(locations)
+        self.location = self._location['name'] 
+        self.gid = self._location['gid']
+
+    @parameterized.expand([('valid_gid', 200), ('invalid_gid', 400)])        
+    def test01_get_free_mac_address(self, case, response_code):
+        """ OVC-001
+        #. Get free mac address, should succeed.
+        #. Get free mac address of invalid grid id, should fail.
+        """        
+        if case == 'valid_gid':
+            gid = self.gid
+        else:
+            gid = self.utils.random_string()
+
+        response = self.api.libcloud.libvirt.getFreeMacAddress(gid=gid)
+        self.assertEqual(response.status_code, response_code)
+
+    @parameterized.expand([('valid_gid', 200), ('invalid_gid', 400)])
+    def test02_get_free_network_id(self, case, response_code):
+        """ OVC-002
+        #. Get free network id, should succeed.
+        #. Get free network id of invalid grid id, should fail.
+        """
+        if case == 'valid_gid':
+            gid = self.gid
+        else:
+            gid = self.utils.random_string()
+
+        response = self.api.libcloud.libvirt.getFreeNetworkId(gid=gid)
+        self.assertEqual(response.status_code, response_code)
+
+    @parameterized.expand([
+            ('valid_newtwork_id', 200), 
+            ('invalid_network_id', 409), 
+            ('invalid_gid', 400),
+            ('invalid_start', 400),
+            ('invalid_end', 400),
+        ])
+    def test03_register_release_network_id_range(self, case, response_code):
+        """ OVC-003
+        #. Create account (AC1), should succeed.
+        #. Create cloudspace (CS1), should succeed.
+        #. Register network id, should succeed.
+        #. Register network id in range of deployed networkids, should fail
+        #. Register network id of invalid grid id, should fail.
+        #. Register network id of invalid start value, should fail.
+        """
+        gid = self.gid
+        start = random.randint(10000, 50000)
+        end = random.randint(10000, 50000)
+
+        if case in ['valid_newtwork_id', 'invalid_network_id']:
+
+            self.lg.info('Create account (AC1), should succeed')
+            account_id = self.api.create_account()
+            self.assertTrue(account_id)
+            self.CLEANUP['accounts'].append(account_id)
+
+            self.lg.info('Create cloudspace (CS1), should succeed')
```

Comment:
Use create_cloudspace method from the framework

Reply:
i did